ebs-nvme-mapping.sh

```diff
@@ -1,10 +1,25 @@
 #!/bin/bash
+# To be used with the udev rule: /etc/udev/rules.d/999-aws-ebs-nvme.rules
 
-PATH="${PATH}:/usr/sbin"
+if [[ ! -x /usr/sbin/nvme ]]; then
+  echo "ERROR: NVME tools not installed." >> /dev/stderr
+  exit 1
+fi
 
-for blkdev in $( nvme list | awk '/^\/dev/ { print $1 }' ) ; do
-  mapping=$(nvme id-ctrl --raw-binary "${blkdev}" | cut -c3073-3104 | tr -s ' ' | sed 's/ $//g')
-  if [[ "${mapping}" == /dev/* ]]; then
-    ( test -b "${blkdev}" && test -L "${mapping}" ) || ln -s "${blkdev}" "${mapping}"
-  fi
-done
+if [[ ! -b ${1} ]]; then
+  echo "ERROR: cannot find block device ${1}" >> /dev/stderr
+  exit 1
+fi
+
+# capture 32 bytes at an offset of 3072 bytes from the raw-binary data
+# not all block devices are extracted with /dev/ prefix
+# use `xvd` prefix instead of `sd`
+# remove all trailing space
+nvme_link=$( \
```

Comment:
In both this block and the starting block of this file, it’s better to rely on the `PATH` and the shell interpreter’s resolution instead of hard-coding the expected path to the binaries in question.

As you cross from Debian/Ubuntu to RHEL/CentOS/Amazon Linux, you will find that the paths to the utilities will change between distributions, and can change even between the same distribution but different releases.

But for the most part, they will remain reachable/executable via the `PATH`.

Reply:
Updated accordingly.